### PR TITLE
CodeLens for #insert section

### LINKED ISSUE #814: CodeLens for #insert section
It is hard to know where {#insert title} is used by another  template which include it. It should be nice to provide codelens which show references and click on codelens should find references.

image

//cc @FroMage

### PR DESCRIPTION
**Pull request by angelozerr**
CodeLens for #insert section

Fixes #814

This PR requires https://github.com/redhat-developer/vscode-quarkus/pull/595 on vscode side to support click on codelens references which must launch the references search.

### CODE REVIEW

**Comment by angelozerr** [comment]
I need to write tests, but the boths PRs should work. Here a demo with renarde todo //cc @FRoMage

QuteCodeLensInsertDemo

**Comment by angelozerr** [comment]
The code is clean and tests exists. Please review it.

**Comment by datho7561** [comment]
oops I was missing the client side PR, my bad

**Review by datho7561 — APPROVED** [review]
Looks pretty good. One small change, but after that it's good to merge. Thanks, Angelo!

File: qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteCodeLens.java [hunk 1]
```diff
@@ -188,17 +193,21 @@ private static void collectFragmentCodeLens(ExtendedDataModelTemplate templateDa
 				cancelChecker);
 	}
 
-	public static void collectInsertCodeLens(QuteProject project, Section section, List<CodeLens> lenses) {
+	public static void collectInsertCodeLens(QuteProject project, Section section, Template template,
```
Comment (JessicaJHee) [inline qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteCodeLens.java:196]: I believe this method can be private?

**Review by JessicaJHee — APPROVED** [review]
Looks good to me and works well!

File: qute.ls/com.redhat.qute.ls/src/test/java/com/redhat/qute/services/codelens/QuteCodeLensForInsertSectionTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
+/*******************************************************************************
+* Copyright (c) 2022 Red Hat Inc. and others.
```
[the remark this answers is not in the document]
Comment (angelozerr) [reply qute.ls/com.redhat.qute.ls/src/test/java/com/redhat/qute/services/codelens/QuteCodeLensForInsertSectionTest.java]: fixed

**Pull request merged**